db-engine/src/main/java/com/splicemachine/db/iapi/util/RowIdUtil.java

```diff
@@ -0,0 +1,33 @@
+package com.splicemachine.db.iapi.util;
+
+import com.splicemachine.db.iapi.error.StandardException;
+import com.splicemachine.db.iapi.types.*;
+import com.splicemachine.primitives.Bytes;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+
+import static com.splicemachine.uuid.Snowflake.TIMESTAMP_MASK;
+import static com.splicemachine.uuid.Snowflake.TIMESTAMP_SHIFT;
+
+public interface RowIdUtil {
+
+    static String toHBaseEscaped(String s) {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < s.length(); i += 2) {
+            sb.append("\\x").append(s, i, i+2);
+        }
+        return sb.toString();
+    }
+
+    static DateTimeDataValue toInstant(DataValueDescriptor s) throws StandardException {
+        if(s == null) {
+            return null;
+        }
+        String hex = toHBaseEscaped(s.getString());
+        long value = Bytes.toLong(Bytes.toBytesBinary(hex));
+        long ts = System.currentTimeMillis() & (~TIMESTAMP_MASK);
+        ts |= (value >> TIMESTAMP_SHIFT) & TIMESTAMP_MASK;
+        return new SQLTimestamp(new Timestamp(ts));
```

Comment:
this could be a static function. then you could have a unit test for toLongTimestamp

```
public static long toLongTimestamp( String str, long time )
{
	String hex = toHBaseEscaped( str );
    long value = Bytes.toLong(Bytes.toBytesBinary(hex));
    long ts = time & (~TIMESTAMP_MASK);
    ts |= (value >> TIMESTAMP_SHIFT) & TIMESTAMP_MASK;
    return ts;
}

static DateTimeDataValue toInstant(DataValueDescriptor s) throws StandardException {
    if(s == null) {
        return null;
    }

    long time = System.currentTimeMillis();
    return new SQLTimestamp(new Timestamp( toLongTimestamp(s.getString(), time)));
}
```